src/lib/useBlockWindowClose.js

```diff
@@ -15,3 +15,19 @@ export default function useBlockWindowClose() {
     };
   });
 }
+
+export function useConditionalBlockWindowClose(condition) {
```

Comment:
This doesn't keep me from going to a different page within Bridge though. Is that something we can prevent?

Reply:
I'd need to add lock/unlock functionality to the router